classes/class-date-interval.php

```diff
@@ -43,86 +37,89 @@ public function get_predefined_intervals() {
-					'start' => Carbon::today( $timezone )->subDays( 30 ),
-					'end'   => Carbon::today( $timezone ),
+					'start' => $today->modify( '-30 days' ),
+					'end'   => $today,
 				),
 				'this-month'     => array(
 					'label' => esc_html__( 'This Month', 'stream' ),
-					'start' => Carbon::today( $timezone )->startOfMonth(),
-					'end'   => Carbon::today( $timezone )->endOfMonth(),
+					'start' => $today->modify( 'first day of this month' ),
+					'end'   => $today->modify( 'last day of this month' )->modify( '+1 day -1 microsecond' ),
 				),
 				'last-month'     => array(
 					'label' => esc_html__( 'Last Month', 'stream' ),
-					'start' => Carbon::today( $timezone )->startOfMonth()->subMonth(),
-					'end'   => Carbon::today( $timezone )->startOfMonth()->subSecond(),
+					'start' => $today->modify( 'first day of last month' ),
+					'end'   => $today->modify( 'last day of last month' )->modify( '+1 day -1 microsecond' ),
 				),
 				'last-3-months'  => array(
 					/* translators: %d: number of months (e.g. "3") */
 					'label' => sprintf( esc_html__( 'Last %d Months', 'stream' ), 3 ),
-					'start' => Carbon::today( $timezone )->subMonths( 3 ),
-					'end'   => Carbon::today( $timezone ),
+					'start' => $today->modify( '-3 months' ),
+					'end'   => $today,
 				),
 				'last-6-months'  => array(
 					/* translators: %d: number of months (e.g. "3") */
 					'label' => sprintf( esc_html__( 'Last %d Months', 'stream' ), 6 ),
-					'start' => Carbon::today( $timezone )->subMonths( 6 ),
-					'end'   => Carbon::today( $timezone ),
+					'start' => $today->modify( '-6 months' ),
+					'end'   => $today,
 				),
 				'last-12-months' => array(
 					/* translators: %d: number of months (e.g. "3") */
 					'label' => sprintf( esc_html__( 'Last %d Months', 'stream' ), 12 ),
-					'start' => Carbon::today( $timezone )->subMonths( 12 ),
-					'end'   => Carbon::today( $timezone ),
+					'start' => $today->modify( '-12 months' ),
+					'end'   => $today,
 				),
 				'this-year'      => array(
 					'label' => esc_html__( 'This Year', 'stream' ),
-					'start' => Carbon::today( $timezone )->startOfYear(),
-					'end'   => Carbon::today( $timezone )->endOfYear(),
+					'start' => $today->modify( 'first day of January' ),
+					'end'   => $today->modify( 'last day of December' )->modify( '+1 day -1 microsecond' ),
 				),
 				'last-year'      => array(
 					'label' => esc_html__( 'Last Year', 'stream' ),
-					'start' => Carbon::today( $timezone )->startOfYear()->subYear(),
-					'end'   => Carbon::today( $timezone )->startOfYear()->subSecond(),
+					'start' => $today->modify( 'first day of January' )->modify( '-1 year' ),
+					'end'   => $today->modify( 'first day of January' )->modify( '-1 microsecond' ),
 				),
-			),
-			$timezone
-		);
+			);
+		} catch ( \Exception $e ) {
+			$date_intervals = array();
+		}
+
+		return apply_filters( 'wp_stream_predefined_date_intervals', $date_intervals, $timezone );
```

Comment:
Definitely 👍 

Since this PR was meant to mainly make the plugin compliant with the latest PHP versions, I glanced over other aspects. However, I think that it makes sense to add docblock comments at least to the code parts that are being modified at a given time (and maybe issue a separate PR later on that would add the comments to other places).